docs/project-versioning.md

```diff
@@ -2,13 +2,13 @@
 redirect_from: "/Project-Versioning"
 ---
 
-# Project Versioning
+## Project Versioning
 
 We are using the semantic versioning standard:
 
 http://semver.org/spec/v2.0.0.html
 
-For instance, we can have: **1.5.1** that corresponds to: **MAJOR.MINOR.PATCH**
+For instance, we can have: **1.5.1-dev.1** that corresponds to: **MAJOR.MINOR.PATCH-PRERELEASE**
```

Comment:
Do we use semantic versioning for API changes such as breaking changes to our module hooks too? The reason I ask is because I don't think I've seen this happen. For example, recently a ton of arena hooks were changed that broke plugins like arena 1v1. I don't believe there was a corresponding major version bump

Reply:
We release a major version once in a while. But we work almost on master (which uses the prerelease versioning), we also have stable branches (e.g. 3.0), that are not going to receive any breaking change and we only apply security fixes if really needed, but they are not used because everyone just wants latest features.

So, when we work with master, every pre-release can add breaking changes, therefore the module authors should upgrade them.
This changelog feature is not to avoid releasing breaking changes but, instead, to let the module authors/contributors be aware that they have to upgrade by also providing the guide on how to do it in a single file (the changelog).

Reply:
Having a changelog full of documented breaking changes is definitely a good thing, I know it'll help me out. I suppose the reason I ran into that breaking change was just because I was using a dev version, so I was sacrificing a that guarantee to make sure I had the latest bug fixes. 

With that in mind though, do you think that the maybe the reason people tend not to use the stable builds is that most of the changes the dev builds include are bug fixes? Maybe that's wrong, but that's what I've seen in my limited time using AC. I use the dev builds because they look like they'll be *more* stable than the 'stable' builds. Regardless of how development or versioning is done, my ideal situation would be to get just the bug fixes, and not breaking changes like changing/removal of core hooks and/or major refactors. I don't want to overstep here, but have you tried merging all low risk bugfixes to the 'stable' branch by default, and merging any other changes to `master`?

Reply:
1. People tend to use the dev version (master branch) because it's more
fixed than the stable version but less stable because introduces breaking
changes and every new commit is potentially decreasing the stability.

2. We can't just import fixes into the previous versions because most of
our fixes also include breaking changes. It's because AC is not yet a
finalized in terms of features and some of them are not properly developed
so they must be refactorized, otherwise you can't provide the proper fix. This kind of approach produces breaking changes but also is prone to generate code conflicts if we want to keep updated both the master and the previous version with content fixing.

We eventually figured out that the semver approach cannot be followed
strictly in our case. Therefore our MAJOR version can be considered more a
"milestone" for now, while the status of AC itself is still in Dev. But
maybe in future things can change and we are using semver for that
eventuality.

Reply:
Oh wow, I didn't know so many fixes included breaking changes :(

Not that this solves any of that, but as a minor side note for #2, I was thinking more of targeting the stable branch first, then merging it to 'master', so that you at least have to resolve conflicts on the development branch instead of a stable branch